tools/test/build_api/build_api_test.py

```diff
@@ -238,5 +240,35 @@ def test_build_library_no_app_config(self, mock_prepare_toolchain, mock_exists,
         self.assertEqual(args[1]['app_config'], None,
                          "prepare_toolchain was called with an incorrect app_config")
 
+    @patch('tools.build_api.intelhex_offset')
+    def test_merge_region_no_fit(self, mock_intelhex_offset):
+        """
+        Test that merge region fails as expected when part size overflows region size.
+        """
+        max_addr = 87444 
+        # create a dummy hex file with above max_addr
+        mock_intelhex_offset.return_value = IntelHex({0:2, max_addr:0})
+
+        self.assertEqual(0, mock_intelhex_offset.return_value.minaddr())
+        self.assertEqual(max_addr, mock_intelhex_offset.return_value.maxaddr())
+    
+        # create application and post-application regions and merge.
+        region_application = Region("application", 10000, 86000, True, "random.hex")
+        region_post_application = Region("postapplication", 100000, 90000, False, None)
+
+        notify = MockNotifier()
+        region_list = [region_application, region_post_application]
+        toolchain = prepare_toolchain(self.src_paths, self.build_path, self.target,
```

Comment:
toolchain needed to pass to merge_region_list()

Reply:
> `merge_region_list(region_list, res, notify, toolchain.config)`

Nope. It needs a config object, or stand in. That's not a toolchain object.

Reply:
Thanks @theotherjimmy got it. Updated the patch.